tf2onnx/rewriter/gemm_rewriter.py

```diff
@@ -0,0 +1,204 @@
+            c_edge_name = _find_edges_name_btw_nodes(inputC_node, add_node)
+
+            alpha = match.get_op("alpha").get_tensor_value()
+
+            gemm = g.make_node("Gemm", inputs=[a_edge_name, b_edge_name, c_edge_name],
+                               attr={"alpha": alpha},
+                               shapes=[g.get_shape(add_node.output[0])],
+                               dtypes=[g.get_dtype(add_node.output[0])])  # add_node.output_shapes[0]
+
+            ops.remove(add_node)
+            ops.remove(matmul_node)
+            ops.remove(mul1_node)
+
+    if patternID==2:    #there is no alpha
+        for match in match_results:
+            # output nodes:
+            add_node = match.get_op('add')
+
+            matmul_node = match.get_op("matmul")
+            mul2_node = match.get_op("mul2")
+            # there is no mul1
+
+            inputA_node = match.get_op("A")
+            inputB_node = match.get_op("B")
+            inputC_node = match.get_op("C")
+
+            # for edges:
+            a_edge_name = _find_edges_name_btw_nodes(inputA_node, matmul_node)
+            b_edge_name = _find_edges_name_btw_nodes(inputB_node, matmul_node)
+            c_edge_name = _find_edges_name_btw_nodes(inputC_node, mul2_node)
+
+            beta = match.get_op("beta").get_tensor_value()
+
+            gemm = g.make_node("Gemm", inputs=[a_edge_name, b_edge_name, c_edge_name], attr={"beta": beta},
+                               shapes=[g.get_shape(add_node.output[0])], dtypes=[g.get_dtype(add_node.output[0])]) # add_node.output_shapes[0]
+
+            ops.remove(add_node)
+            ops.remove(matmul_node)
+            ops.remove(mul2_node)
+
+    if patternID==3:    #there are no alpha and beta
+        for match in match_results:
+            # output nodes:
+            add_node = match.get_op('add')
+
+            matmul_node = match.get_op("matmul")
+            # there are no mul1 and mul2
+
+            inputA_node = match.get_op("A")
+            inputB_node = match.get_op("B")
+            inputC_node = match.get_op("C")
+
+            # for edges:
+            a_edge_name = _find_edges_name_btw_nodes(inputA_node, matmul_node)
+            b_edge_name = _find_edges_name_btw_nodes(inputB_node, matmul_node)
+            c_edge_name = _find_edges_name_btw_nodes(inputC_node, add_node)
+
+            # alpha = match.get_op("alpha").get_tensor_value()
+            # beta = match.get_op("beta").get_tensor_value()
+
+            gemm = g.make_node("Gemm", inputs=[a_edge_name, b_edge_name, c_edge_name],
+                               shapes=[g.get_shape(add_node.output[0])], dtypes=[g.get_dtype(add_node.output[0])]) # add_node.output_shapes[0]
+
+            ops.remove(add_node)
+            ops.remove(matmul_node)
+
+    ops.append(gemm)
+    g.replace_all_inputs(ops, add_node.output[0], gemm.output[0])
+
+    return ops
+
+def _find_edges_name_btw_nodes(sender, sinker):
```

Comment:
_find_edge_name_btw_nodes instead of _find_edges_name_btw_nodes

Reply:
Thanks for your comment. I think the name can be directly obtained by the input and output nodes. In modified codes, the function is deleted.